File: LeetCode/20. Valid Parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        mp = {")":"(", "]":"[", "}":"{"}
        stack = []

        for brace in s:
            if brace in mp:
                if stack and stack[-1] == mp[brace]:
                    stack.pop()
                else: 
                    return False
            else:
                stack.append(brace)
        
        return True if not stack else False

File: LeetCode/20. Valid Parentheses.go
func isValid(s string) bool {
    var stack []rune
    mp := map[rune]rune{
        ')':'(',
        ']':'[',
        '}':'{',
    }

    for _, el := range s {
        if _, ok := mp[el]; ok {
            if len(stack) != 0 && stack[len(stack)-1] == mp[el] {
                stack = stack[:len(stack)-1]
            } else {
                return false
            }
        } else {
            stack = append(stack, el)
        }
    }

    return len(stack) == 0
}
